### PR TITLE
modules: use `bot.say()`'s `trailing` parameter

### PR DESCRIPTION
**Pull request by dgw**
### Description
This updates every core plugin I could find with an existing handler for "too long" output (`instagram`, `reddit`, `tld`, `wiktionary`, & `wikipedia`) to use `bot.say()`'s new `trailing` parameter from #1958.

Two things that _might_ be worth amending, if others think my thought process or assumptions were wrong:

* `instagram`: I rearranged the output to put the caption (most likely to be over-length) at the end. A subsequent commit to this branch (to `wikipedia`) showed a neat hack that would allow leaving the caption where it was, in the middle of the line, but it seems to make the most sense at the end anyway.
* `reddit`: I did _not_ change submission output to use `trailing`, because:
  * The title is the very first thing in the output
  * The submission's link might also be very long, and we can't truncate those
  * If the link is long enough, it plus the rest of the metadata could be longer than allowed anyway, even without the title
  * Basically, there were too many edge cases I could see, so I didn't touch the current post/submission behavior

### Checklist
- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

File: sopel/modules/wikipedia.py [hunk 1]
```diff
@@ -193,7 +200,7 @@ def say_section(bot, trigger, server, query, section):
         return
 
     msg = '{} - {} | "{}"'.format(page_name, section.replace('_', ' '), snippet)
-    bot.say(msg)
+    bot.say(msg, trailing=' [...]"')
```
Comment (Exirel) [inline sopel/modules/wikipedia.py:203]: Why not this bad boy here too?

```suggestion
    bot.say(msg, trailing=' […]"')
```

**Review by Exirel — APPROVED** [review]

File: sopel/modules/wikipedia.py (hunk 1 above)
Comment (dgw) [reply sopel/modules/wikipedia.py:203]: Oh, I was matching what it appended before ([`wikipedia.py` L236](https://github.com/sopel-irc/sopel/pull/1995/files#diff-c5ee1fef33221c1015f3a3313648113e1be501376eeea2354c013db3926958c2L236)), but there's no real reason to do so and I already changed the ellipsis style in this PR's `wiktionary.py` patch… I'm just silly? 😁

Comment (Exirel) [reply sopel/modules/wikipedia.py]: That's why I already approved the PR, I knew you would fix it or have a good reason. ❤️
